src/react/databrowser/objects/details/ObjectRestorationButtonAndModal.tsx

```diff
@@ -0,0 +1,166 @@
+import { ChangeEvent, useState } from 'react';
+import {
+  Banner,
+  Icon,
+  Modal,
+  PrettyBytes,
+  Stack,
+  Text,
+  Wrap,
+} from '@scality/core-ui';
+import { Button, Input } from '@scality/core-ui/dist/next';
+import { useMutation } from 'react-query';
+import { useSelector } from 'react-redux';
+import { AWSError } from 'aws-sdk';
+import { getClients } from '../../../utils/actions';
+import { AppState } from '../../../../types/state';
+import { formatShortDate } from '../../../utils';
+
+type RestoreObjectType = {
+  bucketName: string;
+  objectKey: string;
+  objectLastModifiesOn: string;
+  objectSize: number;
+  objectStorageClass: string;
+  isObjectRestoredOrOnGoing: boolean;
+  objectVersionId?: string;
+};
```

Comment:
nit: I'm wondering, shouldn't we just pass the whole object entity `(s3Object: ObjectEntity)`, the `ObjectEntity` is defined in `src/types/s3.ts`

Reply:
I will pass the entire objectMetadata then. The first thought on my side was to ease the testing. But then I found we already have the mocked OBJECT_METADATA.